From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 15 Jun 2014 21:44:43 +0100
Subject: dmaengine: Avoid ABI change in 3.14.6
Forwarded: not-needed

dmaengine_unmap_data::map_cnt was added to the beginning of the
structure.  Move it after bidi_cnt, where there was previously at
least one byte of padding, and hide it from genksyms.

--- a/include/linux/dmaengine.h
+++ b/include/linux/dmaengine.h
@@ -433,10 +433,12 @@ typedef bool (*dma_filter_fn)(struct dma
 typedef void (*dma_async_tx_callback)(void *dma_async_param);
 
 struct dmaengine_unmap_data {
-	u8 map_cnt;
 	u8 to_cnt;
 	u8 from_cnt;
 	u8 bidi_cnt;
+#ifndef __GENKSYMS__
+	u8 map_cnt;
+#endif
 	struct device *dev;
 	struct kref kref;
 	size_t len;
